okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/AsyncQosIoExceptionHandler.java

```diff
@@ -44,8 +48,19 @@
         return qosIoException.getQosException().accept(new QosException.Visitor<ListenableFuture<Response>>() {
             @Override
             public ListenableFuture<Response> visit(QosException.Throttle exception) {
-                // TODO(rfink): Implement.
-                return Futures.immediateFailedFuture(qosIoException);
+                Optional<Duration> duration = exception.getRetryAfter();
```

Comment:
@chrisalice @j-baker do you think this backoff logic is reasonable?

Reply:
for an obvious style thing, I'd try to do something more like

```java
Optional<Duration> backoff = exception.getRetryAfter().isPresent() ? exception.getRetryAfter() : backoffStrategy.nextBackOff()
```

and then only have one nested if.

However, I think I'm going to have to spend some more time grokking the general picture.

Reply:
done